### ISSUE TITLE
KafkaBinderConfigurationProperties does not handle dot-named kafka configuration properties

### ISSUE THREAD

**Issue by joelmarty** [spring-cloud/spring-cloud-stream-binder-kafka#129]
I recently updated spring-cloud-stream-binder-kafka to 1.2.0 from 1.1.2 using release train Camden.SR1 and somehow my application.yml that looked like this:

```yaml
spring.cloud.stream.kafka.binder:
  brokers: ${vcap.services.MH.credentials.kafka_brokers_sasl}
  configuration:
    security.protocol: SASL_SSL
    sasl.mechanism: PLAIN
    ssl.protocol: TLSv1.2
    ssl.enabled.protocols: TLSv1.2
    ssl.endpoint.identification.algorithm: HTTPS
```
did not work anymore. While debugging, I noticed that the field `configuration` in KafkaBinderConfigurationProperties did not contain the entries 'security.protocol' as expected but an entry 'security', containing a map with an entry 'protocol'. I had to change the configuration to the following to make it work:

```yaml
spring.cloud.stream.kafka.binder:
  brokers: ${vcap.services.MH.credentials.kafka_brokers_sasl}
  configuration:
    "[security.protocol]": SASL_SSL
    "[sasl.mechanism]": PLAIN
    "[ssl.protocol]": TLSv1.2
    "[ssl.enabled.protocols]": TLSv1.2
    "[ssl.endpoint.identification.algorithm]": HTTPS
```

Is it something that is expected ? I don't see how the version number change alone would have caused such behavior change.

I am using spring-boot-starter-parent 1.5.2.RELEASE.
I did not specified any specific spring cloud release train before but since my kafka service is based on version 0.10, I had to manually depend on spring-integration-kafka 2.1.0.RELEASE and kafka-clients 0.10.0.1, if this is in any way relevant.

**Comment by ngawor** [comment]
I am experiencing the same with 1.3.1 (also using kafka client 0.10.0.1).

**Comment by ngawor** [comment]
This is what my working application.properties ended up looking like:

spring.cloud.stream.kafka.binder.configuration.[security.protocol]=SASL_SSL
spring.cloud.stream.kafka.binder.configuration.[sasl.mechanism]=PLAIN
spring.cloud.stream.kafka.binder.configuration.[ssl.protocol]=TLSv1.2
spring.cloud.stream.kafka.binder.configuration.[ssl.enabled.protocols]=TLSv1.2
spring.cloud.stream.kafka.binder.configuration.[ssl.endpoint.identification.algorithm]=HTTPS

**Comment by mbogoevici** [comment]
Closed via https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/commit/001cc5901e4b8d179237c8d12e13f28e2f011e2e

**Comment by zbuske** [comment]
Does anyone have any idea when the fix will be released? The workaround doesn't work for me.

**Comment by sabbyanandan** [comment]
Hi, @zbuske. This PR was merged on the 1.3.x release line and it has been GA'd since early this month. See the [release blog](https://spring.io/blog/2017/10/06/spring-cloud-stream-1-3-goes-ga) for more details.

**Comment by zbuske** [comment]
Thanks for the answer. I am having trouble using Ditmars.Release. It
doesn't seem to be in Maven Central yet and I am having dependency issues
with other Spring packages if I don't use the bom. Could you tell me what
repository I can get it from? I think I'm missing something.

Thanks

Zoa
